### PR TITLE
Remove no-jdk distributions

### LINKED ISSUE #65109: Remove no-jdk distributions for 8.0 release
When we started bundled a JDK into Elasticsearch with 7.0 we also introduced a "no-jdk" distribution for folks that had no intention of using the included JDK. In hindsight this added complexity and confusion around our distribution variants for little benefit. We only provide no-jdk variants of our _archive_ distributions, for packages (rpm/deb) and Docker we only publish artifacts with an included JDK. For the release of 8.0 we want to unify this and remove the no-jdk altogether so that all Elasticsearch distributions will include a bundled JDK.

There are a few items to consider:

- [ ] Remove generation of of no-jdk artifacts from the build.
- [x] Remove publishing of no-jdk artifacts from release manager.
- [x] Update download webpage to no longer link to no-jdk distributions and remove no-jdk page.
- [x] Remove any documentation specifically related to no-jdk distribution (if any).

### PR DESCRIPTION
**Pull request by mark-vieira**
Starting with Elasticsearch 8.0 we will no longer build separate "no-jdk" distributions. All distributions now include by default a bundled JDK and users can configure an alternative JDK distribution via `ES_JAVA_HOME`.

I've left some references to no-jdk distribution handling code to make future backports easier but all the build scripts under `distribution/` have had conditional jdk logic stripped out. Additionally,  there was a lot of lingering code related to OSS distributions which no longer exist which has been removed as well and will be separately backported to `7.x`.

Closes https://github.com/elastic/elasticsearch/issues/65109

### CODE REVIEW

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows

**Review by pugnascotia — APPROVED** [review]

**Review by breskeby — COMMENTED** [review]
nice cleanup! lgtm

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/DistroTestPlugin.java [hunk 1]
```diff
@@ -364,38 +364,16 @@ private static void configureVMWrapperTasks(
...
-                        if (architecture == Architecture.AARCH64) {
-                            continue;
-                        }
-                        // All our Docker images include a bundled JDK so it doesn't make sense to test without one.
-                        if (type.isDocker()) {
-                            continue;
-                        }
-                    }
-                    currentDistros.add(
-                        createDistro(distributions, architecture, type, null, bundledJdk, VersionProperties.getElasticsearch())
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/DistroTestPlugin.java:380]: we can probably remove the bundleJdk parameter from the `createDistro`

**Review by breskeby — COMMENTED** [review]
lgtm

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/DistroTestPlugin.java (hunk 1 above)
Comment (mark-vieira) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/DistroTestPlugin.java:380]: I purposefully didn't to make backports easier. That might be wishful thinking though.

**Pull request merged**